src/NLog/LayoutRenderers/AllEventPropertiesLayoutRenderer.cs

```diff
@@ -135,33 +160,43 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         /// <summary>
         /// The names of caller information attributes.
         /// https://msdn.microsoft.com/en-us/library/hh534540.aspx
+        /// TODO NLog ver. 5 - Remove these properties
         /// </summary>
-        private static HashSet<string> CallerInformationAttributeNames = new HashSet<string>
+        private static List<string> CallerInformationAttributeNames = new List<string>
```

Comment:
why no hashset? Afaik it's more efficient with 3+ elements, and it's static

Reply:
Well I don't need the HashSet-lookup, and the struct-enumerator on List is allocation-free.